### DISCUSSION TITLE
Keeping the grain boundaries intact using the G=gmshGeo(grains) function of MTEX2Gmsh toolbox

### DISCUSSION THREAD

**Discussion by MohammadJoudivand** [mtex-toolbox/mtex#537] [Ask Anything] [answered]
Dear participants
I use Mtex toolbox to create my geometry and post processing it (discritizing and applying in FEM-based software).
I have a problem in saving my geometry and need your help in this regard.
My geometry is a square grid, I use `G=gmshGeo(grains)` to create the geometry package and then save it, but when I use it the initial configuration transforms into a collection of B-splines instead of straight lines.
Are you able to kindly let me know how I should attempt this so that my grains boundaries remain intact and keep their initial forms? In other terms I only want to save geometry as it is not the splined form of it.
Waiting for your reply
Mohammad Hasan Joudivand

**Comment by DorianDepriester** [comment, marked as answer]
Hello Mohammad,
Currently, this feature is currently not available in this toolbox (I can try to add it ASAP). As a workaround, you can export the geometry in a ``.geo`` file:

    savegeo(G,'geometry.geo')

Then, using a text file editor (e.g. notepad++), replace all ``BSpline`` entries by ``Line``. Finally, open this file with Gmsh (in command line or from GUI) to mesh the geometry.

⚠️ **Caveat:** When meshing polygonal lines, Gmsh uses each node of the lines as mesh nodes. Thus, the element sizes are constrained by the node-to-node distance in the polygonal lines. This can result in a huge number of elements. Maybe you should have a look on the [``simplify``](https://doriandepriester.github.io/MTEX2Gmsh/html/improvegeometry.html#5) method.

Hope this helps,
regards.

**Reply by MohammadJoudivand** [reply]
Dear Dorian
Thank you very much
It worked appropriately
Sincerely yours
Mohammad
